reference/multigrid/pmis_kernels.cpp

```diff
@@ -45,7 +84,34 @@ void compute_strong_dep(std::shared_ptr<const DefaultExecutor> exec,
                         remove_complex<ValueType> strength_threshold,
                         matrix::SparsityCsr<ValueType, IndexType>* strong_dep)
 {
-    GKO_NOT_IMPLEMENTED;
+    std::vector<remove_complex<ValueType>> max_values(csr->get_size()[1]);
+
+    for (IndexType col = 0; col < csr->get_size()[1]; ++col) {
+        remove_complex<ValueType> max_val = 0;
+        for (IndexType row = 0; row < csr->get_size()[0]; ++row) {
+            for (auto idx = csr->get_const_row_ptrs()[row]; idx < csr->get_const_row_ptrs()[row + 1]; ++idx) {
+                if (csr->get_const_col_idxs()[idx] == col && row != col) {
+                    max_val = std::max(max_val, gko::abs(csr->get_const_values()[idx]));
+                }
+            }
+        }
+        max_values[col] = max_val;
+    }
+
+std::vector<IndexType> row_offsets(csr->get_size()[0]);
+for (IndexType row = 0; row < csr->get_size()[0]; ++row) {
+    row_offsets[row] = strong_dep->get_row_ptrs()[row];
+}
```

Comment:
it is unnecessary.
In your code, you only need to get it on betweb line 107 and 108 as temporary variable and work on it.